test/smt/smt-proxy.test.ts

```diff
@@ -0,0 +1,126 @@
+import { poseidonContract } from "circomlibjs";
+import { expect } from "chai";
+import { ethers, upgrades } from "hardhat";
+import { deployContracts, publishState } from "../deploy-utils";
+
+const issuerStateTransitions = [
+  require("../mtp/data/issuer_state_transition.json"),
+  require("../mtp/data/issuer_next_state_transition.json"),
+];
+
+describe("State SMT integration tests", function () {
+  let state: any, smt: any;
+
+  beforeEach(async () => {
+    const contracts = await deployContracts(false);
+    state = contracts.state;
+    smt = contracts.smt;
+  });
+
+  it("should proxy upgrade works", async () => {
```

Comment:
This `"it"`is more related to a research of the Proxy functionality rather than a test of our contract.
Consider moving this to `scripts/upgradeContract/upgradeSmtExample.ts` or something

Reply:
I would leave this code in the test because by running CI we can be sure that the contract mock still has relevant changes with the actual contract. It helps to keep the codebase up to date. Your thoughts?

Reply:
Let's move to a separate test file or script (for now) as discussed.